### ISSUE TITLE
[API Proposal]: Add JsonSerializer.SerializeAsync returning string

### ISSUE THREAD

**Issue by Dreamescaper** [dotnet/runtime#107860]
### Background and motivation

JsonSerializer requires async execution for some types, like IAsyncEnumerable. I would be nice to have a method SerializeAsync, which returns serialized string instead of writing to stream.

### API Proposal

```csharp
public static Task<string> SerializeAsync<TValue>(TValue value, JsonSerializerOptions? options = null);
```

### API Usage

```csharp
IAsyncEnumerable<MyType> asyncEnumerable = GetAsyncEnumerable();
var json = await JsonSerializer.SerializeAsync(asyncEnumerable);
```

**Comment by eiriktsarpalis** [comment]
Seems reasonable given that async serialization is a prerequisite for IAsyncEnumerable support. For completeness we should also consider accelerators for all the other supported encoding mediums such as `SerializeToNodeAsync`, `SerializeToUtf8Async` and `SerializeToElementAsync`, `SerializeToDocumentAsync`, etc.

**Comment by Dreamescaper** [comment]
@eiriktsarpalis 
Would it be fine for me to contribute it?

**Comment by eiriktsarpalis** [comment]
The API proposal needs a bit of work before we can consider it for [API review](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md). Bare minimum it should [scope out all the relevant overloads](https://github.com/dotnet/runtime/issues/107860#issuecomment-2352763422) that need to be included as part of this change. It's not something our team can prioritize right now, but if you want to expedite the process you can try by updating the OP to something that can be easily reviewed. You can take a look at [previously approved proposals](https://github.com/dotnet/runtime/issues?q=is%3Aissue+is%3Aopen+label%3Aapi-approved+) for examples of what this should look like.

**Comment by Dreamescaper** [comment]
@eiriktsarpalis 
Would you prefer for me to add  SerializeToNodeAsync, SerializeToUtf8Async and SerializeToElementAsync, SerializeToDocumentAsync, etc. to the proposal? 

Or focus on SerializeAsync, and tackle others separately?

**Comment by eiriktsarpalis** [comment]
I think they should be addressed in tandem. They logically belong to the same set of APIs but they use different names out of disambiguation concerns.

**Comment by Dreamescaper** [comment]
@eiriktsarpalis 
I have updated the proposal.
Currently there are 5 overloads for each of Serialize* methods. I have included Async counterparts for all of them (although I am not sure whether all of them are needed here).
